### PR TITLE
chore(env): set the required Falcon envvars in conftest.py

### LINKED ISSUE #1652: Improve test readme. Without setting the proper env variable 300+ failure occur
If I follow the readme to run the test

> You can manually test changes to the Falcon framework by switching to the directory of the cloned repo and then running pytest:

> $ cd falcon
> $ pip install -r requirements/tests
> $ pytest tests

I get the following result 
` 261 failed, 1819 passed, 20 skipped, 40 warnings, 128 errors`

I've tried with tox and it works correctly.

I've traced the issue to the missing env variable `FALCON_ASGI_WRAP_NON_COROUTINES=Y`

I think the readme should mention it

### PR DESCRIPTION
**Pull request by vytas7**
Closes #1652

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -28,6 +38,10 @@ def pytest_configure(config):
         plugin.mypy_argv.append('--ignore-missing-imports')
 
 
+def pytest_sessionstart(session):
+    os.environ.update(_FALCON_TEST_ENV)
```
Comment (CaselIT) [inline tests/conftest.py:42]: I think `setdefault`  would be better, since it allows controlling the variables from the extern

**Review by CaselIT — CHANGES_REQUESTED** [review]
Thanks for taking this on

File: tests/conftest.py (hunk 1 above)
Comment (vytas7) [reply tests/conftest.py]: sure

**Review by nZac — APPROVED** [review]
:shipit:

**Pull request merged**